### PR TITLE
Improved circuit ports and unitests

### PR DESCRIPTION
**Pull request by Alberto-DM**
- increased unit test coverage for Circuit and Nexxim.
- small fix in get_source_pin_name. 
- renamed create_iport in create_interface_port. 
- renamed create_3dlayout_subcircuit in add_subcircuit_3dlayout. 
- refactored set_sim_option_on_hfss_subcircuit and set_sim_solution_on_hfss_subcircuit. 
- fixed return values on some Nexxim methods. 
- deprecated assign_sin_excitation2ports. 
- added assign_voltage_sinusoidal_excitation_to_ports. 
- added assign_current_sinusoidal_excitation_to_ports. 
- added assign_power_sinusoidal_excitation_to_ports.

### CODE REVIEW

**Review by maxcapodi78 — APPROVED** [review]
LGTM. Great that we add UT and we move methods in right position

File: pyaedt/modeler/PrimitivesCircuit.py [hunk 1]
```diff
@@ -136,7 +137,18 @@ def create_wire(self, points_array):
 
     @aedt_exception_handler
     def create_iport(self, name, posx=0.1, posy=0.1, angle=0):
-        """Create a port.
+        """Create an interface port.
+
+        .. deprecated:: 0.4.0
+           Use :func:`Circuit.modeler.components.create_interface_port` instead.
```
Comment (akaszynski) [inline pyaedt/modeler/PrimitivesCircuit.py:143]: Good work here.

**Review by akaszynski — APPROVED** [review]
Nice work!

File: pyaedt/modeler/PrimitivesCircuit.py (hunk 1 above)
Comment (Alberto-DM) [reply pyaedt/modeler/PrimitivesCircuit.py:143]: Thanks.

**Pull request merged**